NumPy main fails to compile with gcc 13.1 on fedora 37

Gcc 13.1 compiles the maintenance/1.24.x branch, but fails with main. The problem file looks to be `numpy/core/src/npymath/halffloat.cpp`. There is no error message, the compilation just fails and the build exits, no other files are built after that. Note that gcc 12 compiles the main branch with no problems. I'm looking for suggestions on how to track this down, in particular, how to get more information from the failure.

build error:

```Bash
In file included from numpy/core/src/common/common.hpp:9,
                 from numpy/core/src/npymath/halffloat.cpp:12:
numpy/core/src/common/npstd.hpp:23:12: error: ‘uint8_t’ has not been declared in ‘std’
   23 | using std::uint8_t;
      |            ^~~~~~~
numpy/core/src/common/npstd.hpp:24:12: error: ‘int8_t’ has not been declared in ‘std’
   24 | using std::int8_t;
      |            ^~~~~~
numpy/core/src/common/npstd.hpp:25:12: error: ‘uint16_t’ has not been declared in ‘std’
   25 | using std::uint16_t;
      |            ^~~~~~~~
numpy/core/src/common/npstd.hpp:26:12: error: ‘int16_t’ has not been declared in ‘std’
   26 | using std::int16_t;
      |            ^~~~~~~
numpy/core/src/common/npstd.hpp:27:12: error: ‘uint32_t’ has not been declared in ‘std’
   27 | using std::uint32_t;
      |            ^~~~~~~~
numpy/core/src/common/npstd.hpp:28:12: error: ‘int32_t’ has not been declared in ‘std’
   28 | using std::int32_t;
      |            ^~~~~~~
numpy/core/src/common/npstd.hpp:29:12: error: ‘uint64_t’ has not been declared in ‘std’
   29 | using std::uint64_t;
      |            ^~~~~~~~
numpy/core/src/common/npstd.hpp:30:12: error: ‘int64_t’ has not been declared in ‘std’
   30 | using std::int64_t;
      |            ^~~~~~~
numpy/core/src/common/npstd.hpp:31:12: error: ‘uintptr_t’ has not been declared in ‘std’
   31 | using std::uintptr_t;
      |            ^~~~~~~~~
numpy/core/src/common/npstd.hpp:32:12: error: ‘intptr_t’ has not been declared in ‘std’
   32 | using std::intptr_t;
      |            ^~~~~~~~
numpy/core/src/common/npstd.hpp:34:12: error: ‘uint_fast16_t’ has not been declared in ‘std’
   34 | using std::uint_fast16_t;
      |            ^~~~~~~~~~~~~
numpy/core/src/common/npstd.hpp:35:12
```

--- a/numpy/core/src/common/half.hpp
+++ b/numpy/core/src/common/half.hpp
@@ -72,7 +72,7 @@ class Half final {
     {
     #if defined(NPY_HAVE_AVX512FP16)
         __m128d md = _mm_load_sd(&f);
-        bits_ = static_cast<uint16_t>(_mm_cvtsi128_si32(_mm_castph_si128(_mm_cvtpd_ph(mf))));
+        bits_ = static_cast<uint16_t>(_mm_cvtsi128_si32(_mm_castph_si128(_mm_cvtpd_ph(md))));
     #elif defined(NPY_HAVE_VSX3) && defined(NPY_HAVE_VSX_ASM)
         __vector double vf64 = vec_splats(f);
         __vector unsigned short vf16;

--- a/numpy/core/src/common/npstd.hpp
+++ b/numpy/core/src/common/npstd.hpp
@@ -4,6 +4,7 @@
 #include <cstddef>
 #include <cstring>
 #include <cctype>
+#include <cstdint>
 
 #include <string>
 #include <algorithm>

